artview/core/common.py

```diff
@@ -160,9 +164,12 @@ def display_data(self, data):
             the table """
 
         if len(data)==0:
-            data = ["No data for selected ROI."]
+            data = [["No data for selected ROI."]]
+            nrows = 0
+            ncols = 0
+        else:
+            nrows, ncols = data.shape[0], data.shape[1]
```

Comment:
1. Yes (if you mean outside the radar region: the circle region not the square window)
2. yes if radar is not None, otherwise data will be empty